packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,248 @@
+
+    const accountDeployed = await isAccountDeployed(this.network, address);
+
+    if (!accountDeployed) {
+      const { callData } = getAccContractAddressAndCallData(
+        this.account.publicKey,
+      );
+
+      const deployAccountpayload = {
+        classHash: ACCOUNT_CLASS_HASH,
+        contractAddress: address,
+        constructorCalldata: callData,
+        addressSalt: this.account.publicKey,
+      };
+
+      invocations.unshift({
+        type: TransactionType.DEPLOY_ACCOUNT,
+        payload: deployAccountpayload,
+      });
+    }
+
+    if (
+      // Get Starknet expected not to show the confirm dialog, therefore,
+      // `enableAuthorize` will set to false to bypass the confirmation
+      // TODO: enableAuthorize should set default to true
+      enableAuthorize &&
+      !(await this.getExecuteTxnConsensus(
+        address,
+        calls,
+        abis,
+        details,
+        accountDeployed,
+      ))
+    ) {
+      throw new UserRejectedRequestError() as unknown as Error;
+    }
+
+    if (!accountDeployed) {
+      await createAccount(
+        this.network,
+        this.account.publicKey,
+        this.account.privateKey,
+        CAIRO_VERSION,
+        true,
+      );
+    }
+
+    const estimateFeeResp = await getEstimatedFees(
+      this.network,
+      address,
+      this.account.privateKey,
+      this.account.publicKey,
+      invocations as unknown as Invocations,
+      details?.version ?? TRANSACTION_VERSION,
+      !accountDeployed,
+    );
+
+    const resp = await executeTxnUtil(
+      this.network,
+      address,
+      this.account.privateKey,
+      calls,
+      abis,
+      {
+        nonce: accountDeployed ? undefined : 1,
+        maxFee: estimateFeeResp.suggestedMaxFee,
+      },
+    );
+
+    if (resp.transaction_hash) {
+      const callData = calls[0].calldata as Calldata;
+
+      const txn: Transaction = {
+        txnHash: resp.transaction_hash,
+        txnType: VoyagerTransactionType.INVOKE,
+        chainId: this.network.chainId,
+        senderAddress: address,
+        contractAddress: calls[0].contractAddress,
+        contractFuncName: calls[0].entrypoint,
+        contractCallData: callData.map((data: string) => {
+          try {
+            return `0x${BigInt(data).toString(16)}`;
+          } catch (error) {
+            // data is already send to chain, hence we should not throw error
+            return '0x0';
+          }
+        }),
+        finalityStatus: TransactionStatus.RECEIVED,
+        executionStatus: TransactionStatus.RECEIVED,
+        status: '', // DEPRECATED LATER
+        failureReason: '',
+        eventIds: [],
+        timestamp: Math.floor(Date.now() / 1000),
+      };
+
+      const stateManager = new TransactionStateManager();
+      await stateManager.addTransaction(txn);
+    }
+
+    return resp;
+  }
+
+  protected async getExecuteTxnConsensus(
+    address: string,
+    calls,
+    abis,
+    details,
```

Comment:
types has to delcare

Reply:
i will say we may need to display the dialog in that way (below logic is similar to getSendTxnText in snapUtils)

First of all: we always need to show:
- Display Signer Address
- Display account deployment required (similar to UI estimate fee)
- For each call data (support multiple call data), we display:
   - Display Contract
   - Display Call Data
   - if it is ERC20 token and contract Function Name === 'transfer', then we display:
     - Display Sender Address
     - Display Recipient Address
     - Display Amount
- Display Estimated Gas Fee(ETH/STRK)
- Display Network
- Display deploy account transaction

Reply:
<img width="355" alt="Capture d’écran 2024-08-28 à 21 16 40" src="https://github.com/user-attachments/assets/b7f4f673-6594-4693-8cc5-7c7f162954b3">

Looks like this wdyt ?